packages/api-providers/src/ext-providers/polkadot/polkadot-provider.ts

```diff
@@ -250,17 +256,30 @@ export class PolkadotProvider extends EventBus<ExtensionProviderEvents> {
 
     logger.info('Extensions', extensions);
 
-    // Check for extensions length to insure polkadot extension is installed
-    if (extensions.length === 0) {
-      logger.warn("Polkadot extension isn't installed");
-      throw WebbError.from(WebbErrorCodes.PolkaDotExtensionNotInstalled);
+    // Check whether the extension is existed or not
+    const currentExtension = extensions.find((ex) => ex.name === wallet.name);
+    if (!currentExtension) {
+      logger.warn(`${wallet.title} extension isn't installed`);
+
+      switch (wallet.id) {
+        case WalletId.Polkadot: {
+          throw WebbError.from(WebbErrorCodes.PolkaDotExtensionNotInstalled);
+        }
+
+        case WalletId.Talisman: {
+          throw WebbError.from(WebbErrorCodes.TalismanExtensionNotInstalled);
+        }
+
+        default: {
+          throw new Error('Unknown wallet');
```

Comment:
Can we have a typed error here   **EX** `throw WebbError.from(WebbErrorCodes.UnkownWallet)`